### PR TITLE
Render 5xx server errors.

### PR DESCRIPTION
**Pull request by andorp**
Fixes: #121

### CODE REVIEW

File: src/Frontend/Middleware.hs [hunk 1]
```diff
@@ -28,18 +28,23 @@ import Frontend.Core
 import Frontend.Prelude
 
 
-catch404 :: Bool -> Middleware
-catch404 devMode app req cont = app req $ \resp -> cont $ f resp
+catchErrors :: Bool -> Middleware
+catchErrors devMode app req cont = app req $ \resp -> cont $ f resp
```
Comment (fisx) [inline src/Frontend/Middleware.hs]: `catchHttpErrors`?  `catch4xx5xx`?

File: src/Frontend/Core.hs [hunk 2]
```diff
@@ -420,18 +420,25 @@ instance Page CsrfToken where
     -- token. However since this is just used for testing purposes so far there is no need to allow
     -- that.
 
-data Page404 = Page404
+data ServerErrorPage = Page404 | Page5xx
```
Comment (fisx) [inline src/Frontend/Core.hs]: `HttpErrorPage`?  Could be renamed together with `catchErrors` for consistency, but i don't insist.

File: src/Frontend/Core.hs [hunk 3]
```diff
@@ -420,18 +420,25 @@ instance Page CsrfToken where
...
+        h1_ [class_ "main-heading"] errorTitle
+        h2_ [class_ "sub-header"] errorDesc
         div_ $ a_ [class_ "btn-cta", href_ P.login] "Start"
+      where
+        errorTitle = case p of
+            Page404 -> "404"
+            Page5xx -> "500"
+        errorDesc = case p of
+            Page404 -> "Diese Seite gibt es leider nicht."
+            Page5xx -> "Interner Serverfehler"
```
Comment (fisx) [inline src/Frontend/Core.hs]: `.` missing?

File: src/Frontend/Middleware.hs [hunk 4]
```diff
@@ -28,18 +28,23 @@ import Frontend.Core
...
   where
     f :: Response -> Response
-    f resp = if statusCode status /= 404
-        then resp
-        else responseBuilder status headers builder
+    f resp = case statusCode status of
+                404          -> responseBuilder status headers body404
```
Comment (fisx) [inline src/Frontend/Middleware.hs]: should we also catch the rest of `4xx` here?

**Comment by fisx** [comment]
LGTM.  Lots of nit-picks, please check if you have an opinion.  You can fix them yourself, argue about them, or give this PR to me and I'll do it myself.  :-)

**Comment by fisx** [comment]
@andorp ok?

**Comment by andorp** [comment]
LGTM

**Pull request merged**